src/lib/Libwin/passwd.c

```diff
@@ -4814,20 +4815,21 @@ logon_pw(char *username,
 	size_t credl,
 	int (*decrypt_func)(char *, int, size_t, char **),
 	int use_winsta,
-	char msg[4096])
+	char* msg)
 {
-	static  char	*id = "logon_pw";
-	int		i;
-	struct	passwd *pwdp;
-	SID	*usid;
-	char	*homedir;
-	char  	thepass[PWLEN+1];
-	char 	realname[UNLEN+1];
-	char	domain[PBS_MAXHOSTNAME+1] = "";
-	char	*p;
-	char	msg2[4096];
+	int		i = 0;
+	struct	passwd *pwdp; = NULL
+	SID	*usid = NULL;
+	char	*homedir = NULL;
+	char  	thepass[PWLEN+1] = {'\0'};
+	char 	realname[UNLEN+1] = {'\0'};
+	char	domain[PBS_MAXHOSTNAME+1] = {'\0'};
+	char	*p = {'\0'};
+	char	msg2[MSGLEN] = {'\0'};
```

Comment:
The problem with adding a macro that only gets used in one place is that you never know the length of the array that is actually being passed. The logon_pw() function is called in four places. In src/resmom_win/mom_stage_file.c log_buffer is passed where LOG_BUF_SIZE is defined as 4096. In src/tools/pbs_runas_win.c msg is passed where CMDLINE_LEN is defined as 4096. It's a coincidence that these values are all defined to be 4096. If one of them changes, we could end up with memory corruption that could be very hard to diagnose.

If you are relying on the array to be of a specific length, then every caller should use the same macro to define its size. Go ahead and reuse LOG_BUF_SIZE if you like, but just make sure it's consistent.

The use of strcat() in this function confuses me. At line 4832, we ensure msg is an empty string. Why do we sprintf() to msg2 and then strcat() it to msg? Why not just sprintf(), or better yet snprintf(), to msg directly?